elixir/apps/domain/lib/domain/gateways.ex

```diff
@@ -23,6 +23,12 @@ defmodule Domain.Gateways do
     |> Repo.aggregate(:count)
   end
 
+  def fetch_gateway_by_id(id) do
```

Comment:
Nothing wrong here, just wanted to put a quick comment that it might be nice if we could come up with some convention for functions like these.  Where this function is being used doesn't really pose any security risk, but if this function gets used else where we always need to be certain that the `id` being passed in isn't coming from a potentially end-user controlled origin.  That's why passing in the `subject` for almost all of our other functions is good, because even if the end-user can control what `id` is passed in it won't return data they shouldn't see, because it will be scoped by their subject (i.e. their account).

Reply:
I agree, there unfortunately is no subject in this context though. It would be nice to have stronger conventions agreed, i.e. split context modules

Reply:
Yeah, totally understand why this is needed.  Just wanted to start a conversation to see if it sparked any ideas around ways to make stuff like this more obviously dangerous when someone reaches for this function in the future.